core/src/main/java/de/danoeh/antennapod/core/storage/DBReader.java

```diff
@@ -927,82 +927,142 @@ public static FeedMedia getFeedMedia(final long mediaId) {
+                } else {
+                    feedPlayedTimeCountAll += media.getPosition() / 1000;
+                }
+
                 if (media.getPlaybackCompletionDate() != null || media.getPlayedDuration() > 0) {
                     episodesStarted++;
                 }
+
+                if (item.isPlayed() || media.getPosition() != 0) {
+                    episodesStartedIncludingMarked++;
+                }
+
                 feedTotalTime += media.getDuration() / 1000;
                 episodes++;
             }
             feedTime.add(new StatisticsItem(
-                    feed, feedTotalTime, feedPlayedTime, episodes, episodesStarted));
+                    feed, feedTotalTime, feedPlayedTime, feedPlayedTimeCountAll, episodes,
+                    episodesStarted, episodesStartedIncludingMarked));
             totalTime += feedPlayedTime;
+            totalTimeCountAll += feedPlayedTimeCountAll;
         }
 
-        Collections.sort(feedTime, (item1, item2) -> {
-            if(item1.timePlayed > item2.timePlayed) {
-                return -1;
-            } else if(item1.timePlayed < item2.timePlayed) {
-                return 1;
-            } else {
-                return 0;
-            }
-        });
+        if (sortByCountAll) {
+            Collections.sort(feedTime, (item1, item2) ->
+                    compareLong(item1.timePlayedCountAll, item2.timePlayedCountAll));
+        } else {
+            Collections.sort(feedTime, (item1, item2) ->
+                    compareLong(item1.timePlayed, item2.timePlayed));
+        }
 
         adapter.close();
-        return new StatisticsData(totalTime, feedTime);
+        return new StatisticsData(totalTime, totalTimeCountAll, feedTime);
+    }
+
+    /**
+     * Compares two {@code long} values. Long.compare() is not available before API 19
+     *
+     * @return 0 if long1 = long2, less than 0 if long1 &lt; long2,
+     * and greater than 0 if long1 &gt; long2.
+     */
+    private static int compareLong(long long1, long long2) {
+        if (long1 > long2) {
+            return -1;
+        } else if (long1 < long2) {
+            return 1;
+        } else {
+            return 0;
+        }
     }
 
     public static class StatisticsData {
+        /**
+         * Simply sums up time of podcasts that are marked as played
+         */
+        public long totalTimeCountAll;
+
+        /**
+         * Respects speed, listening twice, ...
+         */
         public long totalTime;
+
         public List<StatisticsItem> feedTime;
 
-        public StatisticsData(long totalTime, List<StatisticsItem> feedTime) {
+        public StatisticsData(long totalTime, long totalTimeCountAll, List<StatisticsItem> feedTime) {
             this.totalTime = totalTime;
+            this.totalTimeCountAll = totalTimeCountAll;
             this.feedTime = feedTime;
         }
     }
 
     public static class StatisticsItem {
         public Feed feed;
         public long time;
+
```

Comment:
not sure we need all these comments
I prefer good variable names to usually outdated comments